File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
import java.util.*;
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
       int low = 0, high = 0, sum = 0, result = Integer.MAX_VALUE;

       while(high < nums.length){
            sum += nums[high];
            while(sum>=target){
                int length = high - low + 1;
                result = Math.min(result, length);
                sum -= nums[low];
                low++;
            }
            high++;
       }
       if(result == Integer.MAX_VALUE){
        return 0;
       }else{
        return result;
       }
    }
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        low=0
        high=0
        result=float('inf')
        sum = 0
        while high<len(nums):
            sum = sum+nums[high]
            while sum >= target:
                length = high-low+1
                result=min(result,length)
                sum = sum - nums[low]
                low+=1
            high+=1
        if result == float('inf'):
            return 0
        else:
            return result